app/scripts/services/room.js

```diff
@@ -3,9 +3,9 @@
     var ref = firebase.database().ref().child("rooms");
     var rooms = $firebaseArray(ref);
       
-    this.addRoom = function(newRoom) {
-        rooms.$add(newRoom)
-    }  
+    this.addChatRoom = function(newChatRoomName) {
+        rooms.$add({$value: newChatRoomName});
```

Comment:
in here, `newChatRoomName` will the the value you passed from the controller. In your $add call, you want to pass an object with the property names matching the columns in your database. Since we are using `name` as the column that stores the attribute holding a room's name, we'll use that.

``` javascript
//code
  rooms.$add({
    name: newChatRoomName;
  });
//code
```